packages/x-charts/src/ChartsYAxis/ChartsYAxis.tsx

```diff
@@ -141,25 +141,28 @@ function ChartsYAxis(inProps: ChartsYAxisProps) {
     // - No data is associated to the axis (other scale types)
     return null;
   }
+
   return (
     <AxisRoot
       transform={`translate(${position === 'right' ? left + width : left}, 0)`}
       className={classes.root}
     >
       {!disableLine && (
-        <Line
-          y1={yScale.range()[0]}
-          y2={yScale.range()[1]}
-          className={classes.line}
-          {...slotProps?.axisLine}
-        />
+        <Line y1={top - 1} y2={top + height} className={classes.line} {...slotProps?.axisLine} />
```

Comment:
Why `-1`?
```suggestion
        <Line y1={top} y2={top + height} className={classes.line} {...slotProps?.axisLine} />
```

Reply:
Visual fix, not sure why it happens though. 

Gotta zoom in on the top left corner though

Without -1
![Screenshot 2024-07-10 at 16 14 28](https://github.com/mui/mui-x/assets/9891233/d4737dcd-a12d-48fe-9fc9-466df605436b)

With -1
![Screenshot 2024-07-10 at 16 14 35](https://github.com/mui/mui-x/assets/9891233/adf07716-d5b0-4c4d-a420-057ed2f16b08)

Reply:
Ho yes, we have the same on the x/y axis shared bixel

But `stroke-linecap` could be more reliable

https://developer.mozilla.org/fr/docs/Web/SVG/Attribute/stroke-linecap